src/components/Tooltip.js

```diff
@@ -0,0 +1,135 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Animated, Text, View,
+} from 'react-native';
+import Hoverable from './Hoverable';
+import getTooltipStyles from '../styles/getTooltipStyles';
+
+const propTypes = {
+    // The text to display in the tooltip.
+    text: PropTypes.string.isRequired,
+
+    // Child to wrap with Tooltip. Must be only child.
+    children: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.element,
+        PropTypes.instanceOf(Component),
+    ]).isRequired,
+};
+
+class Tooltip extends Component {
+    constructor(props) {
+        super(props);
+
+        this.animation = new Animated.Value(0);
+
+        // The wrapper view containing the wrapped content along with the Tooltip itself.
+        this.wrapperView = null;
+        this.width = 0;
+        this.height = 0;
+
+        // The tooltip (popover) itself.
+        this.tooltip = null;
+        this.tooltipWidth = 0;
+        this.tooltipHeight = 0;
+
+        // The distance between the left side of the rendered view and the left side of the window
+        this.xOffset = 0;
+
+        // The distance between the top of the rendered view and the top of the window
+        this.yOffset = 0;
+
+        this.getPosition = this.getPosition.bind(this);
+        this.showTooltip = this.showTooltip.bind(this);
+        this.hideTooltip = this.hideTooltip.bind(this);
+    }
+
+    /**
+     * Get the position and measure the size of the wrapper view and the tooltip itself.
+     */
+    getPosition() {
+        this.wrapperView.measureInWindow((x, y, width, height) => {
+            this.xOffset = x;
+            this.yOffset = y;
+            this.width = width;
+            this.height = height;
+        });
+        this.tooltip.measureInWindow((x, y, width, height) => {
+            this.tooltipWidth = width;
+            this.tooltipHeight = height;
+        });
+    }
+
+    /**
+     * Display the tooltip in an animation.
+     */
+    showTooltip() {
+        Animated.timing(this.animation, {
+            toValue: 1,
+            duration: 140,
+        }).start();
+    }
+
+    /**
+     * Hide the tooltip in an animation.
+     */
+    hideTooltip() {
+        Animated.timing(this.animation, {
+            toValue: 0,
+            duration: 140,
+        }).start();
+    }
+
+    render() {
+        const interpolatedSize = this.animation.interpolate({
+            inputRange: [0, 1],
+            outputRange: [0, 1],
+        });
```

Comment:
Just wondering why do we need the `interpolate`? Not super familiar with what it is doing and was hoping you could help me understand.

Reply:
Yeah, I'll do my best to give a hand-wavy explain, but it's really only to the best of my understanding. I bet @Luke9389 could explain maybe better (given his background in teaching math, as this is really just linear algebra mumbo jumbo).

Basically `interpolate` changes a number gradually over the set duration of the animation. It defines how to map the inputs provided for an animation to the output result of the animation, over the duration of the animation.

You could provide different parameters for the input range and output range, but this maps `0 -> 0`, `0.25 -> 0.25`, etc... basically something like this:

- At time 0%, the interpolation factor will be 0
- At time 50%, the interpolation factor will be 0.5
- At time 100%, the interpolation factor will be 1

You can think of that as the "identity function" of interpolation. Whereas if we did this:

```
this.animation.interpolate({
    inputRange: [0, 1],
    outputRange: [0, 2],
});
```

We would end up with:

- At time 0%, the interpolation factor will be 0.
- At time 50%, the interpolation factor will be 1
- At time 100%, the interpolation factor will be 2.

`showTooltip` triggers the animation to begin with its starting value of 0 and end with a value of 1, so we end up with something like:

- At time 0%, the animation value of 0 is interpolated to 0 so the size is 0.
- At time 50%, the animation value of 0.5 is interpolated to 0.5, so the size is 0.5.
- At time 100%, the animation value of 1 is interpolated to 1, so the size is 1.

`hideTooltip` triggers the animation  to begin with its starting value of 1 and end with a value of 0, so we end up with something like:

- At time 0%, the animation value of 1 is interpolated to 1, so the size is 1.
- At time 50%, the animation value of 0.5 is interpolated to 0.5, so the size is 0.5.
- At time 100%, the animation value of 0 is interpolated to 0, so the size is 0.

I think that's mostly correct. Hopefully I didn't just confuse you more 😅

Reply:
Ok, can we just animate our value from `0` to `2`?

Reply:
And in this case we are mapping the input to the same output range?

Reply:
Here's a [link to the docs](https://reactnative.dev/docs/animations#interpolation) on animation interpolation. I'm not sure that we need it for `showTooltip` (bc our input is the same as our output), but I think we would still need it for `hideTooltip`.

Reply:
Thanks for the link @Luke9389! Maybe I'm still missing it. Why would we need it for `hideTooltip`?

Reply:
@marcaaron Is there some concern you have about this implementation? What alternative would you suggest?

Reply:
Well I'm just trying to understand why this is here first... but my concern is that this seems unnecessary. As an alternative, I'd suggest to use the animated value directly instead of the interpolated value.

Reply:
You were right, the interpolation was unnecessary. It's removed now